core/src/main/java/nl/nn/adapterframework/webcontrol/api/SendJmsMessage.java

```diff
@@ -64,14 +65,20 @@ public Response putJmsMessage(MultipartBody inputDataMap) throws ApiException {
 		}
 
 		String fileEncoding = resolveTypeFromMap(inputDataMap, "encoding", String.class, StreamUtil.DEFAULT_INPUT_STREAM_ENCODING);
-		String jmsRealm = resolveStringFromMap(inputDataMap, "realm");
+		String connectionFactory = resolveStringFromMap(inputDataMap, "connectionFactory");
 		String destinationName = resolveStringFromMap(inputDataMap, "destination");
 		String destinationType = resolveStringFromMap(inputDataMap, "type");
 		String replyTo = resolveTypeFromMap(inputDataMap, "replyTo", String.class, "");
 		boolean persistent = resolveTypeFromMap(inputDataMap, "persistent", boolean.class, false);
 		boolean synchronous = resolveTypeFromMap(inputDataMap, "synchronous", boolean.class, false);
+		boolean lookupDestination = resolveTypeFromMap(inputDataMap, "lookupDestination", boolean.class, false);
 
-		JmsSender qms = jmsBuilder(jmsRealm, destinationName, persistent, destinationType, replyTo, synchronous);
+		String jmsRealm = JmsRealmFactory.getInstance().findJmsRealm(connectionFactory);
+		if(jmsRealm == null) {
+			throw new ApiException("connection factory ["+connectionFactory+"] does not have a realm");
+		}
```

Comment:
You should not convert the connectionFactory back to a realm and use setJmsRealm(), but used connectionFactory directly